### PR TITLE
Update the full ParameterHandler example.

### PR DESCRIPTION
**Pull request by drwells**
This commit fixes several small issues, such as the path to a deal.II header file, code formatting, and the format of the printed output.

### CODE REVIEW

File: include/deal.II/base/parameter_handler.h [hunk 1]
```diff
@@ -1176,212 +1176,236 @@ namespace Patterns
...
+ *                        "Laplace",
+ *                        Patterns::Anything(),
+ *                        "String identifying the equation we want to solve");
+ *     prm.declare_entry ("Equation 2",
+ *                        "Elasticity",
+ *                        Patterns::Anything());
+ *
+ *     // declare parameters for the first equation
+ *     prm.enter_subsection ("Equation 1 Settings");
+ *     prm.declare_entry ("Matrix type",
```
Comment (bangerth) [inline include/deal.II/base/parameter_handler.h]: We've more recently started to use a style where we enclose things that are in a subsection into braces, i.e.,
```
  prm.enter_subsection (...);
  {
    prm.declare_entry (...);
  }
  prm.leave_subsection;
```
This has the advantage that it also visually indicates what is logically happening. Would you mind updating the example this way?

File: include/deal.II/base/parameter_handler.h [hunk 2]
```diff
@@ -1176,212 +1176,236 @@ namespace Patterns
...
- *     }
+ *   void Problem::get_parameters (ParameterHandler &prm)
+ *   {
+ *     // entries of the problem class
+ *     outfile = prm.get ("Output file");
+ *     std::string equation1 = prm.get ("Equation 1"),
+ *                 equation2 = prm.get ("Equation 2");
+ *     // get parameters for the first equation
+ *     prm.enter_subsection ("Equation 1 Settings");
+ *     matrix1 = prm.get ("Matrix type");
```
Comment (bangerth) [inline include/deal.II/base/parameter_handler.h]: same here, enclose in braces

**Comment by drwells** [comment]
Good point. I updated all of them to use extra braces.

**Comment by bangerth** [comment]
Great. Thanks for cleaning up my 20 year old mess!

**Comment by drwells** [comment]
Not a problem. The only substantial issue was the incorrect header path (I wouldn't call it a mess).

**Pull request merged**